scripts/Packages/Package_Gullwing__QFP_SOIC_SO/Readme.md

```diff
@@ -0,0 +1,151 @@
+  - Custom naming (`custom_name_format`) {python format string}
+    - The full default format string is '{man:s}_{mpn:s}_{pkg:s}-{pincount:d}-1EP_{size_x:g}x{size_y:g}mm_P{pitch:g}mm{suffix:s}_EP{ep_size_x:g}x{ep_size_y:g}mm_Mask{mask_size_x:g}x{mask_size_y:g}mm{suffix2:s}{vias:s}' (The same parameters can be used in your custom format. Exposed pad parameters are not available for components without exposed pad.)
+
+_Note: Contributions intended for the official library shall not include the manufacturer or part number unless the footprint is specific to that manufacturer or part. Similarly avoid custom naming for official library contributions unless required to achieve the requested name (Example TI-specific naming)._
+
+### Package dimensions
+![dimension example](./documentation/dimension_system.svg)
+- Body size (`body_size_x`, `body_size_y`, `overall_height`) {dimension}
+- Lead dimensions:
+  - Overall size representing lead tip to lead tip dimension (`overall_size_x`, `overall_size_y`) {dimension}
+  - Lead width (`lead_width`) {dimension}
+  - Lead length representing tip till bend (`lead_len`)
+- lead pitch, currently equal for all sides (`pitch`) {float}
+- pin count (`num_pins_x`, `num_pins_y`) {int}
+  - num_pins_x=0 is used for generating SOIC like packages.
+  - num_pins_y=0 is used to generate SOIC like package footprints but with inverted pin numbering (Mirrored numbering scheme. Some manufactures use this style in their datasheets. Make sure you are not looking at the bottom view before using this. Not supported for QFP and similar.)
+
+### Exposed pad handling:
+![exposed pad example](../documentation/ep_handling.svg)
+- Size of package exposed pad or slug (`EP_size_x`, `EP_size_y`) {dimension}
+- Optional the size of the footprint pad (`EP_size_x_overwrite`, `EP_size_y_overwrite`) {float}
+   - Pad size is equal to nominal package pad size if not given.
+   - Use this to create a soldermask defined pad.
+- Optional the size of the mask cutout (`EP_mask_x`, `EP_mask_y`) {float}
+   - Use to create solder mask defined pads (in combination with `EP_size_x_overwrite`)
+- Paste is typically split into a regular grid with (`EP_num_paste_pads`) {[int (x), int (y)]}
+  - The optional paste coverage multiplier determines how much of the exposed copper area is covered by paste. (`EP_paste_coverage`) {float (0..1), default=0.65}
+
+### Thermal vias
+A package with exposed pad can generate a version with thermal vias. This will be generated in addition to the normal version.
+
+![exposed pad example](../documentation/thermal_vias.svg)
+
+``` yaml
+  thermal_vias:
+    # thermal via version parameters
+```
+- Number of vias generated in the regular grid (`count`) {[int (x), int (y)]}
+- Final hole size (`drill`) {float}
+- Optional grid (`grid`) {[float (x), float (y)]}
+  - Auto generated if not given (outermost pad will touch pad edge.)
```

Comment:
Period should be after the closing paren.

Reply:
done